### PR TITLE
feat(http): make interceptingWriter reimplement common interfaces

### LINKED ISSUE #1092: Support for WebSockets in Server interface
net/http added support for websockets in v1.12 ([discussion](https://github.com/golang/go/issues/26937)). The change however requires applications to use an additional interface ([Hijacker](https://github.com/golang/go/blob/go1.12/src/net/http/server.go#L179)) to 'hijack' the connection and access the I/O channels. When applications use go-kit, the http.ResponseWriter is replaced by an interceptingWriter here: https://github.com/go-kit/kit/blob/v0.10.0/transport/http/server.go#L99. This only [embeds](https://github.com/go-kit/kit/blob/v0.10.0/transport/http/server.go#L228) the original `http.ResponseWriter`. To open up support for websockets, we needs to embed the Hijacker interface as well.

### PR DESCRIPTION
**Pull request by Reasno**
successor to #1093 and closes #1092.

### CODE REVIEW

**Comment by peterbourgon** [comment]
Sure! Tests, please? 😇

**Comment by Reasno** [comment]
> Sure! Tests, please? 😇

Test already included in the commit. The diff is collapsed by GitHub for being too large.

**Comment by peterbourgon** [comment]
Oops! You're right! Sorry!

File: transport/http/intercepting_writer.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+// reimplementInterfaces returns a wrapped version of the embedded ResponseWriter
+// and selectively implements the same combination of additional interfaces as
+// the wrapped one. The interfaces it may implement are: http.Hijacker,
+// http.CloseNotifier, http.Pusher, http.Flusher and io.ReaderFrom. The standard
+// library is known to assert the existence of these interfaces and behaves
+// differently. This implementation is derived from
+// https://github.com/felixge/httpsnoop.
+func (w *interceptingWriter) reimplementInterfaces() http.ResponseWriter {
```
Comment (peterbourgon) [inline transport/http/intercepting_writer.go:34]: Can we make this into a constructor? e.g.

```go
func newInterceptingWriter(w http.ResponseWriter) http.ResponseWriter {
    ...
}
```

**Pull request merged**